src/core/tokenize/util.ts

```diff
@@ -10,7 +10,7 @@ export function resolve (
 	let ret = reference ?? undefined;
 
 	if (typeof ret === 'string') {
-		ret = prism.languageRegistry.getLanguage(ret)?.grammar;
+		ret = prism.languageRegistry.getLanguage(ret)?.resolvedGrammar;
```

Comment:
This is still not fully resolving cases like:

```js
{
  id: "foo",
  require: bar
  grammar: {
    token: {
      ...
      inside: "bar"
    }
  }
}
```

The `bar` language is _imported_ and can be accessed via the `languages` array associated with the `foo` language. However, `languages["bar"]` is evaluated lazily. Until it's accessed for the first time, the `bar` language won't be added to the `languageRegistry`, and `resolve(prism, "bar")` will **always** return `undefined`.

Honestly, I don't know how to solve it without providing some extra info on the grammar we resolve. 🤷‍♂️